File: templates/typescript/computer-use/types/beta.ts
export interface BetaBaseBlock {
  type: string;
  id?: string;
  cache_control?: { type: 'ephemeral' };
}

export interface BetaTextBlock extends BetaBaseBlock {
  type: 'text';
  text: string;
}

export interface BetaImageBlock extends BetaBaseBlock {
  type: 'image';
  source: {
    type: 'base64';
    media_type: 'image/png';
    data: string;
  };
}

export interface BetaToolUseBlock extends BetaBaseBlock {
  type: 'tool_use';
  name: string;
  input: Record<string, any>;
}

export interface BetaThinkingBlock extends BetaBaseBlock {
  type: 'thinking';
  thinking: any;
  signature?: string;
}

export interface BetaToolResultBlock extends BetaBaseBlock {
  type: 'tool_result';
  content: (BetaTextBlock | BetaImageBlock)[] | string;
  tool_use_id: string;
  is_error: boolean;
}

export type BetaContentBlock = BetaTextBlock | BetaImageBlock | BetaToolUseBlock | BetaThinkingBlock | BetaToolResultBlock;

export interface BetaMessageParam {
  role: 'user' | 'assistant';
  content: BetaContentBlock[] | string;
}

export interface BetaMessage {
  content: BetaContentBlock[];
} 

File: templates/python/computer-use/custom_types/beta.py
from dataclasses import dataclass
from typing import Literal, Union, Optional, List, Dict, Any
from typing import TypedDict

@dataclass
class BetaTextBlock:
    type: Literal['text']
    text: str
    id: Optional[str] = None
    cache_control: Optional[Dict[Literal['type'], Literal['ephemeral']]] = None

@dataclass
class BetaImageSource:
    type: Literal['base64']
    media_type: Literal['image/png']
    data: str

@dataclass
class BetaImageBlock:
    type: Literal['image']
    source: BetaImageSource
    id: Optional[str] = None
    cache_control: Optional[Dict[Literal['type'], Literal['ephemeral']]] = None

@dataclass
class BetaToolUseBlock:
    type: Literal['tool_use']
    name: str
    input: Dict[str, Any]  # ActionParams type
    id: Optional[str] = None
    cache_control: Optional[Dict[Literal['type'], Literal['ephemeral']]] = None

class BetaThinkingEnabled(TypedDict):
    type: Literal['enabled']
    budget_tokens: int

class BetaThinkingDisabled(TypedDict):
    type: Literal['disabled']

@dataclass
class BetaThinkingBlock:
    type: Literal['thinking']
    thinking: Union[BetaThinkingEnabled, BetaThinkingDisabled]
    signature: Optional[str] = None
    id: Optional[str] = None
    cache_control: Optional[Dict[Literal['type'], Literal['ephemeral']]] = None

@dataclass
class BetaToolResultBlock:
    type: Literal['tool_result']
    content: Union[List[Union[BetaTextBlock, BetaImageBlock]], str]
    tool_use_id: str
    is_error: bool
    id: Optional[str] = None
    cache_control: Optional[Dict[Literal['type'], Literal['ephemeral']]] = None

# Define BetaLocalContentBlock before BetaMessageParam
BetaLocalContentBlock = Union[BetaTextBlock, BetaImageBlock, BetaToolUseBlock, BetaThinkingBlock, BetaToolResultBlock]

@dataclass
class BetaMessageParam:
    role: Literal['user', 'assistant']
    content: List[BetaLocalContentBlock]
    id: Optional[str] = None